Adding a new limitation to call out typos in radius resource type (#1405)
radius-project/docs: docs/content/reference/limitations.md

As a workaround make sure to use distinct names for both containers and gateways.

### Typos in Radius Resource Type Names

> Deploying a Radius Resources with a typo in the resource type will result in an error being thrown during deployment. For example, when `Application.Core/Extenders` is defined as `Application.Core/Extender` you will get an error messaging similar to:

Reviewer: ```suggestion
Deploying a Radius Resources with a typo in the resource type or an unsupported resource type will result in an Azure provider related error being thrown during deployment. For example, when `Application.Core/Extenders` is defined as `Application.Core/Extender` you will get an error messaging similar to:
```

```
Azure deployment failed, please ensure you have configured an Azure provider with your Radius environment: https://docs.radapp.io/guides/operations/providers/azure-provider/
```

> As a workaround make sure to use correct resource type names while defining radius resources.

Reviewer: ```suggestion
We are working on making this error clearer. In the meantime, when you see this error please make sure to ensure that the resource type name is correctly specified while defining radius resources in your application/environment bicep file.
```

> We are working on making this error clearer. In the meantime, when you see this error please make sure to ensure that the resource type name is correctly specified while defining radius resources in your application/environment bicep file.

Reviewer 2: ```suggestion
We are working on making this error clearer. In the meantime, when you see this error please ensure that the resource type name is correctly specified in your Radius Application or Environment Bicep file.
```
